src/gengowatcher/watcher.py

```
@@ -426,6 +428,27 @@ def _process_new_job(self, job_id, title, reward, url, source):
 
         self.state.save_state()
 
+        # Notify UI that a new job was added, but only if it was stored in state
+        job_in_state = False
+        try:
+            jobs_attr = getattr(self.state, "jobs", None)
+            if isinstance(jobs_attr, dict):
+                job_in_state = job_id in jobs_attr or str(job_id) in jobs_attr
+        except Exception as e:
```

Ruff (0.14.14)

Do not catch blind exception: `Exception`

(BLE001)